.github/workflows/tests_without_binary.yml

```diff
@@ -11,11 +11,17 @@ on:
         required: true
         description: The version of PennyLane to use. Valid values are either 'release' (most recent release candidate), 'stable' (most recent git-tag) or 'latest' (most recent commit from master)
   pull_request:
+    paths:
+      - .github/workflows/tests_without_binary.yml
+      - pennylane_lightning/core/*
```

Comment:
Do we need `core` and `pennylane_lightning/lightning_qubit/` for the no-bin wheel to be testable?

Reply:
`core` contains a few Python base modules so the files therein must be included. I think I'll flip this upside-down and change for 
```
    paths-ignore:
      - pennylane_lightning/core/src/**
```
so run if there are changes other than the C++ source files. This is better for Python tests which could also be affected to changes in `setup.py` for example.